tests/state_migration_test.go

```diff
@@ -55,15 +57,81 @@ func TestMigration_ContinuesRestartAndMigration(t *testing.T) {
 		newPathKey := append([]byte("databaseDirectory"), common.Int64ToByteBigEndian(uint64(database.StateTrieDB))...)
 		newStateTriePath, err := node.ChainDB().GetMiscDB().Get(newPathKey)
 		assert.NoError(t, err)
-		assert.NotEqual(t, stateTriePath, newStateTriePath)
+		assert.NotEqual(t, stateTriePath, newStateTriePath, "migration failed")
 		stateTriePath = newStateTriePath
 	}
 
 	stopNode(t, fullNode)
 }
 
-// if migration status is set on miscDB and a node is restarted, migration should start
+// state trie DB should be determined by the values of miscDB
 func TestMigration_StartMigrationByMiscDB(t *testing.T) {
+	fullNode, cn, validator, _, workspace, _, _, _ := newSimpleBlockchain(t, 10)
+	defer os.RemoveAll(workspace)
+
+	stateTriePathKey := append([]byte("databaseDirectory"), common.Int64ToByteBigEndian(uint64(database.StateTrieDB))...)
+
+	// use the default StateTrie DB if it is not set on miscDB
+	{
+		// check if stateDB value is not stored in miscDB
+		key, err := cn.ChainDB().GetMiscDB().Get(stateTriePathKey)
+		assert.Error(t, err)
+		assert.Len(t, key, 0)
+
+		// write values in stateDB and check if the values are stored in DB
+		entries := writeRandomValueToStateTrieDB(t, cn.ChainDB().NewBatch(database.StateTrieDB))
+		stopNode(t, fullNode) // stop node to release DB lock
+		checkIfStoredInDB(t, filepath.Join(cn.ChainDB().GetDBConfig().Dir, "statetrie"), entries)
+		fullNode, cn = startNode(t, workspace, validator)
+	}
+
+	// use the state trie DB that is specified in miscDB
+	{
+		// change stateDB value in miscDB
+		newDBPath := "NEW_STATE_TRIE_DB_PATH"
+		err := cn.ChainDB().GetMiscDB().Put(stateTriePathKey, []byte(newDBPath))
+		assert.NoError(t, err)
+
+		// an error expected on node start
+		stopNode(t, fullNode)
+		_, _, err = newKlaytnNode(t, workspace, validator)
+		assert.Error(t, err, "start failure expected, changed state trie db has no data")
+	}
+}
+
+func writeRandomValueToStateTrieDB(t *testing.T, batch database.Batch) map[string]string {
+	entries := make(map[string]string, 10)
+
+	for i := 0; i < 10; i++ {
+		key, value := common.MakeRandomBytes(common.HashLength), common.MakeRandomBytes(400)
+		err := batch.Put(key, value)
+		assert.NoError(t, err)
+		entries[string(key)] = string(value)
+	}
+	assert.NoError(t, batch.Write())
+
+	return entries
+}
+
+func checkIfStoredInDB(t *testing.T, dir string, entries map[string]string) {
+	dbs := make([]*leveldb.DB, 4)
```

Comment:
```suggestion
func checkIfStoredInDB(t *testing.T, dir string, numPartiton uint, entries map[string]string) {
	dbs := make([]*leveldb.DB, numPartition)
```
How about getting `DBConfig.NumStateTriePartitions` value instead of using hard-coded value?

Reply:
review reflected